x-pack/plugins/maps/public/classes/layers/choropleth_layer_wizard/create_choropleth_layer_descriptor.ts

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import uuid from 'uuid/v4';
+import {
+  AGG_TYPE,
+  COLOR_MAP_TYPE,
+  FIELD_ORIGIN,
+  SCALING_TYPES,
+  SOURCE_TYPES,
+  STYLE_TYPE,
+  VECTOR_STYLES,
+} from '../../../../common/constants';
+import { getJoinAggKey } from '../../../../common/get_agg_key';
+import {
+  AggDescriptor,
+  ColorDynamicOptions,
+  EMSFileSourceDescriptor,
+  ESSearchSourceDescriptor,
+} from '../../../../common/descriptor_types';
+import { VectorStyle } from '../../styles/vector/vector_style';
+import { VectorLayer } from '../vector_layer/vector_layer';
+import { EMSFileSource } from '../../sources/ems_file_source';
+// @ts-ignore
+import { ESSearchSource } from '../../sources/es_search_source';
+import { getDefaultDynamicProperties } from '../../styles/vector/vector_style_defaults';
+
+const defaultDynamicProperties = getDefaultDynamicProperties();
+
+function createChoroplethLayerDescriptor({
+  sourceDescriptor,
+  leftField,
+  rightIndexPatternId,
+  rightIndexPatternTitle,
+  rightTermField,
+}: {
+  sourceDescriptor: EMSFileSourceDescriptor | ESSearchSourceDescriptor;
+  leftField: string;
+  rightIndexPatternId: string;
+  rightIndexPatternTitle: string;
+  rightTermField: string;
+}) {
+  const metricsDescriptor: AggDescriptor = { type: AGG_TYPE.COUNT };
+  const joinId = uuid();
+  const joinKey = getJoinAggKey({
+    aggType: metricsDescriptor.type,
+    aggFieldName: metricsDescriptor.field ? metricsDescriptor.field : '',
+    rightSourceId: joinId,
+  });
+  return VectorLayer.createDescriptor({
```

Comment:
The default layer-name looks "off".

E.g. it would say `World Countries` while it should really say something like:
- `Count of kibana_sample_data_logs`
- `Count of kibana_sample_Data_logs by World Countries`

Maybe just add a custom `.label`-property here?

Alternatively, maybe we could override `VectorLayer#getDisplayName` and make it really smart and auto-generate a label-name based on if it has a join and styled by that join value?

Reply:
Both of those names are really long. I don't think its a good idea to put both join and source titles in the description as that will lead to really long names that will cause line wrapping and are cumbersome to read. How about `World Countries by count`?

Reply:
Discussed with @nreese. Will leave this for separate PR, and move implementation into `VectorLayer#getDisplayName`